src/cryptography/hazmat/backends/openssl/hashes.py

```diff
@@ -22,8 +22,8 @@ def __init__(self, backend, algorithm, ctx=None):
             ctx = self._backend._ffi.gc(
                 ctx, self._backend._lib.Cryptography_EVP_MD_CTX_free
             )
-            evp_md = self._backend._lib.EVP_get_digestbyname(
-                algorithm.name.encode("ascii"))
+            name = self._backend._build_openssl_digest_name(algorithm)
+            evp_md = self._backend._lib.EVP_get_digestbyname(name)
             if evp_md == self._backend._ffi.NULL:
                 raise UnsupportedAlgorithm(
                     "{0} is not a supported hash on this backend.".format(
```

Comment:
Maybe just use the `name` var, instead of `algorith.name`.

On Sun, Aug 28, 2016 at 9:22 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/hashes.py
> <https://github.com/pyca/cryptography/pull/3116#discussion_r76531882>:
>
> > @@ -22,8 +22,8 @@ def __init__(self, backend, algorithm, ctx=None):
> >              ctx = self._backend._ffi.gc(
> >                  ctx, self._backend._lib.Cryptography_EVP_MD_CTX_free
> >              )
> > -            evp_md = self._backend._lib.EVP_get_digestbyname(
> > -                algorithm.name.encode("ascii"))
> > +            name = self._backend._build_openssl_digest_name(algorithm)
> > +            evp_md = self._backend._lib.EVP_get_digestbyname(name)
> >              if evp_md == self._backend._ffi.NULL:
> >                  raise UnsupportedAlgorithm(
> >                      "{0} is not a supported hash on this backend.".format(
>
> Do you want a separate error or just an expansion to something like "{0}
> with digest size {1} is not a supported hash on this backend"?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyca/cryptography/pull/3116/files/a03ff0fbbe02eb73e6a655520f71ee6fc91cdcac#r76531882>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAADBEzagbAmE4Hl9fmgnD_aNq4A4zvfks5qkYuTgaJpZM4Jur9K>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6